pkg/cmd/variable/list/list.go

```diff
@@ -236,3 +253,20 @@ func populateSelectedRepositoryInformation(client *http.Client, host string, var
 	}
 	return nil
 }
+
+func (v *Variable) ExportData(fields []string) map[string]interface{} {
+	e := reflect.ValueOf(v).Elem()
+	fieldByName := func(val reflect.Value, field string) reflect.Value {
+		return val.FieldByNameFunc(func(s string) bool {
+			return strings.EqualFold(field, s)
+		})
+	}
+	data := map[string]interface{}{}
+
+	for _, f := range fields {
+		sf := fieldByName(e, f)
+		data[f] = sf.Interface()
+	}
+
+	return data
+}
```

Comment:
Think this isn't needed given the default behavior of exporting structs except we will need to add JSON tags to the remaining `Variable` fields for their lowercased versions so they are exported with the appropriate names.

https://github.com/cli/cli/blob/b0da755b2cb89bdace95203c441933c642b8ea92/pkg/cmd/variable/list/list.go#L174-L181

```go
type Variable struct {
	Name             string            `json:"name"`
	Value            string            `json:"value"`
	UpdatedAt        time.Time         `json:"updated_at"`
	Visibility       shared.Visibility `json:"visibility"`
	SelectedReposURL string            `json:"selected_repositories_url"`
	NumSelectedRepos int               `json:"num_selected_repos"`
}
```

Reply:
```suggestion
```

Reply:
It appears that `secret list` command does not filter by argument fields as follows

```shell
# on test repo
$ gh secret list --json name
[
  {
    "name": "HOGE",
    "updated_at": "2024-01-06T00:40:37Z",
    "visibility": "",
    "selected_repositories_url": "",
    "num_selected_repos": 0
  }
]
```

Therefore, I think it is necessary to implement `ExportData` including the `secret list`.

https://github.com/cli/cli/blob/586eb8b865ea539832716a3d8c5cb42b11ee784c/api/export_pr.go#L8

My implementation is inadequate and the snake case may need to be further modified as follows

```golang
func (v *Variable) ExportData(fields []string) map[string]interface{} {
	e := reflect.ValueOf(v).Elem()
	fieldByName := func(val reflect.Value, field string) reflect.Value {
		return val.FieldByNameFunc(func(s string) bool {
			return strings.EqualFold(field, s)
		})
	}
	data := map[string]interface{}{}
	for _, f := range fields {
		switch f {
		case "updated_at":
			data[f] = v.UpdatedAt
		case "selected_repositories_url":
			data[f] = v.SelectedReposURL
		case "num_selected_repos":
			data[f] = v.NumSelectedRepos
		default:
			sf := fieldByName(e, f)
			data[f] = sf.Interface()
		}
	}
	return data
}
```

Reply:
> It appears that secret list command does not filter by argument fields as follows

Oof; sounds like we have some follow up work addressing those behaviors and shortcomings, thanks for explaining 🤗  I want to bring this up with @samcoe for his thoughts as I apparently overlooked the original secret changes referenced.